### PR TITLE
Tls support

### PR DESCRIPTION
**Pull request by DeliciousBounty**
Hello 
This Pull request add support for tls errors according to the issue #94 .
The ignore parameter is passed trough all active checks. 
I changed http_client , req,  macro, additional_checks,..
I am here for any questions :)

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -24,11 +24,12 @@ clap = { version = "4.0.29", features = ["derive"] }
 const_format = "0.2.30"
 serde = { version = "^1.0", features = ["derive"] }
 serde_json = "^1.0"
-cherrybomb-engine = "^0.1"
+cherrybomb-engine = {path = "cherrybomb-engine"}
```
Comment (GuyL99) [inline Cargo.toml:27]: path

File: cherrybomb-engine/Cargo.toml [hunk 2]
```diff
@@ -13,7 +13,7 @@ edition = "2021"
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 
 [dependencies]
-cherrybomb-oas = "^0.1"
+cherrybomb-oas = {path = "../cherrybomb-oas"}
```
Comment (GuyL99) [inline cherrybomb-engine/Cargo.toml:16]: path

File: cherrybomb-engine/src/scan/active/http_client/req.rs [hunk 3]
```diff
@@ -196,7 +203,10 @@ impl AttackRequest {
             .collect()
     }
     pub async fn send_request_with_response(&self) -> (String, bool) {
-        let client = reqwest::Client::new();
+        let client = Client::builder()
+            .danger_accept_invalid_certs(self.ignore_tls_errors)
```
Comment (GuyL99) [inline cherrybomb-engine/src/scan/active/http_client/req.rs:207]: Are you sure that the boolean matches? I mean will it get the desired outcome or is it the opposite?

File: src/main.rs [hunk 4]
```diff
@@ -54,10 +54,12 @@ fn merge_options(conf: &mut Config, opt: &Options) {
         conf.security.push(value.clone());
     }
     if let Some(value) = &opt.ignore_tls_errors {
+        dbg!(&value);
```
Comment (GuyL99) [inline src/main.rs:57]: dbg!@!!

**Review by GuyL99 — CHANGES_REQUESTED** [review]

**Comment by DeliciousBounty** [comment]
I am opening a newer

**Pull request closed without merging**